File: src/data-scratch/data_science_from_scratch/strategies_library/mystrength.py
import json

import pika

from data_science_from_scratch import routing_key
from data_science_from_scratch.library.probability import mystrength


def mystrength_strategy(self, body: dict):  # noqa: E501
    """ signal strength """
    actual = body["actual"]
    expected = body["expected"]
    strength = mystrength(actual, expected)
    out_dict = dict(
        actual=actual,
        expected=expected,
        strength=strength,
        status_code="200"
    )
    self.channel.basic_publish(
        exchange=self.props.reply_to,
        routing_key=routing_key,
        properties=pika.BasicProperties(correlation_id=self.props.correlation_id),
        body=json.dumps(out_dict).encode("utf-8")
    )

File: src/data-scratch-node-library/ts-data-scratch/strategies_library/mystrength.ts
import { getStrength } from "../signal-strength";

/*
//    actual = body["actual"]
//    expected = body["expected"]
//    strength = mystrength(actual, expected)
//    out_dict = dict(
//        actual=actual,
//        expected=expected,
//        strength=strength,
//    )
*/
interface IstrengthInput {
    actual: number;
    expected: number;
}

const strength = (strengthInput: IstrengthInput) => new Promise(
    async (resolve, reject) => {
        try {
            console.log("strength")
            console.log("strengthInput = ", strengthInput)
            var actual = strengthInput.actual;
            var expected = strengthInput.expected;
            var strength = getStrength(actual, expected);
            var payload = {
                "expected": expected,
                "actual": actual,
                "strength": strength
            };
            console.log("payload = ", payload)
            resolve(console.log(payload));
        } catch (e) {
            reject(
                console.log(`unable to resolve getStrength ${e}`)
            );
        }
    },
);
